include/boost/spirit/home/x3/support/traits/container_traits.hpp

```diff
@@ -30,20 +30,28 @@ namespace boost { namespace spirit { namespace x3 { namespace traits
 
     namespace detail
     {
-        BOOST_MPL_HAS_XXX_TRAIT_DEF(value_type)
-        BOOST_MPL_HAS_XXX_TRAIT_DEF(iterator)
-        BOOST_MPL_HAS_XXX_TRAIT_DEF(size_type)
-        BOOST_MPL_HAS_XXX_TRAIT_DEF(reference)
+        BOOST_TTI_HAS_TYPE(value_type)
+        BOOST_TTI_HAS_TYPE(iterator)
+        BOOST_TTI_HAS_TYPE(size_type)
+        BOOST_TTI_HAS_TYPE(reference)
+        BOOST_TTI_HAS_TYPE(key_type)
+        BOOST_TTI_HAS_MEMBER_FUNCTION(reserve)
     }
 
-    template <typename T, typename Enable = void>
-    struct is_container
-      : mpl::bool_<
-            detail::has_value_type<T>::value &&
-            detail::has_iterator<T>::value &&
-            detail::has_size_type<T>::value &&
-            detail::has_reference<T>::value>
-    {};
+    template <typename T>
+    using is_container = std::integral_constant<bool,
+            detail::has_type_value_type<T>::value &&
+            detail::has_type_iterator<T>::value &&
+            detail::has_type_size_type<T>::value &&
+            detail::has_type_reference<T>::value>;
+
+    template <typename T>
+    using is_reservable = std::integral_constant<bool,
+            detail::has_member_function_reserve<T, void, boost::mpl::vector<size_t>>::value>;
+
+    template <typename T>
+    using is_associative = std::integral_constant<bool,
+    		detail::has_type_key_type<T>::value>;
 
```

Comment:
The documentation says BOOST TTI uses boost::mpl::bool_ too ... 
The type you wanted me to remove in our code